### PR TITLE
jellyfin-ffmpeg: fix build

### PR DESCRIPTION
**Pull request by itepastra**
Add a patch that fixes the compile errors from `libavcodec/libxeve.c` by changing the offending lines to those in upstream

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/jellyfin-ffmpeg/default.nix [hunk 1]
```diff
@@ -16,28 +16,34 @@ in
...
+      "--disable-ptx-compression" # https://github.com/jellyfin/jellyfin/issues/7944#issuecomment-1156880067
+    ];
 
-  postPatch = ''
-    for file in $(cat debian/patches/series); do
-      patch -p1 < debian/patches/$file
-    done
+    patches = [
+      # in libxeve.c there was a difference from upstream that broke the build
+      ./xeve-fix.patch
```
Comment (dotlambda) [inline pkgs/development/libraries/jellyfin-ffmpeg/default.nix:30]:
```suggestion
      (fetchpatch2 {
        url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/3e6c7948626f19c46c1a630c788ea6bbd9e7fbcb";
        hash = "";
      })
```

**Comment by jopejoe1** [comment]
Would be redundant with #353198

File: pkgs/development/libraries/jellyfin-ffmpeg/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/libraries/jellyfin-ffmpeg/default.nix]: A comment explaining the patch or a `name` would be good.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 353168` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>immich</li>
    <li>jellyfin</li>
    <li>jellyfin-ffmpeg</li>
    <li>jellyfin-ffmpeg.bin</li>
    <li>jellyfin-ffmpeg.data</li>
    <li>jellyfin-ffmpeg.dev</li>
    <li>jellyfin-ffmpeg.doc</li>
    <li>jellyfin-ffmpeg.lib</li>
    <li>jellyfin-ffmpeg.man</li>
  </ul>
</details>

**Review by dotlambda — APPROVED** [review]

**Review by Atemu — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/353198

**Pull request closed without merging**